### ISSUE TITLE
DOC(dev): `npm install --global` does not work for me

### ISSUE THREAD

**Issue by corneliusroemer** [nextstrain/auspice#1587]
For some reason `npm install --global` errored for me.

On M1 mac, I need to use osx-64 conda environment as the arm channel does not have nodejs=14, only nodejs=16 and that didn't work.

But when I run `npm install --global` I get the following error:

```log
❯ npm install --global .            
npm WARN @babel/eslint-parser@7.16.3 requires a peer of eslint@^7.5.0 || ^8.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb@15.1.0 requires a peer of eslint@^3.19.0 || ^4.3.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-airbnb-base@11.3.2 requires a peer of eslint@^3.19.0 || ^4.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-plugin-jsx-a11y@5.1.1 requires a peer of eslint@^2.10.2 || ^3 || ^4 but none is installed. You must install peer dependencies yourself.
npm WARN react-addons-css-transition-group@15.6.2 requires a peer of react@^15.4.2 but none is installed. You must install peer dependencies yourself.

npm ERR! code ENOENT
npm ERR! syscall rename
npm ERR! path /opt/homebrew/Caskroom/miniforge/base/envs/auspice-dev/lib/node_modules/.staging/auspice-2fbdfe28/node_modules/@babel/code-frame
npm ERR! dest /opt/homebrew/Caskroom/miniforge/base/envs/auspice-dev/lib/node_modules/.staging/@babel/code-frame-236720d6
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, rename '/opt/homebrew/Caskroom/miniforge/base/envs/auspice-dev/lib/node_modules/.staging/auspice-2fbdfe28/node_modules/@babel/code-frame' -> '/opt/homebrew/Caskroom/miniforge/base/envs/auspice-dev/lib/node_modules/.staging/@babel/code-frame-236720d6'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/corneliusromer/.npm/_logs/2022-10-25T14_28_17_066Z-debug.log
```

What do I need to do? What are peer dependencies? I couldn't find how to do this in the dev docs, maybe worth adding for js noobs like me?

**Comment by jameshadfield** [comment]
> On M1 mac, I need to use osx-64 conda environment as the arm channel does not have nodejs=14, only nodejs=16 and that didn't work.

Could you print the commands you're running and I'll test - I presume it's something like

```
conda env create -n something -c conda-forge nodejs=16 # failed ?!? How so?
conda env create -n something -c conda-forge nodejs=14 # worked
conda activate something
npm install --global
# error as above
```

**Comment by tsibley** [comment]
@corneliusroemer Can you upload the log file that npm produces?

    /Users/corneliusromer/.npm/_logs/2022-10-25T14_28_17_066Z-debug.log

> What are peer dependencies?

They're dependencies which are required at runtime but not (or previously not, this changed in some npm version) automatically installed. (There's reasonable reasons for this, but yeah, it's weird.)

In any case, the peer dependency stuff you saw are warnings only and almost certainly unrelated to the error you're seeing.

**Comment by victorlin** [comment]
I've encountered this in the past. I believe it was due to having multiple conda environments with different Node.js versions sharing the same NPM cache (`npm config get cache`).

The immediate solution was to clear the cache (`npm cache clean --force`). To prevent it from recurring, I now set the NPM cache to live within individual Conda environments:

```sh
conda activate environment
npm config set cache "$CONDA_PREFIX"/.npm-cache
```
